rope/base/oi/doa.py

```diff
@@ -114,24 +131,55 @@ class _SocketReceiver(_MessageReceiver):
     def __init__(self):
         self.server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self.data_port = 3037
+        self.key = os.urandom(32)
+
         while self.data_port < 4000:
             try:
-                self.server_socket.bind(('', self.data_port))
+                self.server_socket.bind(('localhost', self.data_port))
                 break
             except socket.error:
                 self.data_port += 1
         self.server_socket.listen(1)
 
     def get_send_info(self):
-        return str(self.data_port)
+        return '%d:%s' % (self.data_port,
```

Comment:
If you were using ``random.SystemRandom()``, you could use its methods, for example:
```python
rd = random.SystemRandom()
rand_str = {:X}'.format(rd.getrandbits(25))
```
with slightly more predictable and manageable results.

Reply:
I don't think you meant this line, but I'm not sure how it's different than `urandom` that I am using now. That seems just as simple/manageable to me for this case.

Reply:
I'm not sure I understand the suggestion here. `random.SystemRandom` is just a wrapper around `urandom` that lets you get other things like random integers, floats, etc. -- but if you just want a random string, `urandom` is easier.